conformance/tests/gateway-dynamic-listeners.go

```diff
@@ -0,0 +1,223 @@
+				}},
+				Conditions: []metav1.Condition{{
+					Type:   string(v1beta1.ListenerConditionAccepted),
+					Status: metav1.ConditionTrue,
+					Reason: "", //any reason
+				}},
+				AttachedRoutes: 1,
+			})
+		}
+
+		for i := 0; i < tlsPortCount; i++ {
+			port := nextPort(portStart, portEnd, ports)
+			hostname := v1beta1.Hostname(fmt.Sprintf("%v.example.com", port))
+
+			listeners = append(listeners, v1beta1.Listener{
+				Name:     v1beta1.SectionName(strconv.Itoa(port)),
+				Port:     v1beta1.PortNumber(port),
+				Hostname: &hostname,
+				Protocol: v1beta1.HTTPSProtocolType,
+				AllowedRoutes: &v1beta1.AllowedRoutes{
+					Namespaces: &v1beta1.RouteNamespaces{From: &same},
+				},
+				TLS: &v1beta1.GatewayTLSConfig{
+					CertificateRefs: []v1beta1.SecretObjectReference{{
+						Name: v1beta1.ObjectName(certNN.Name),
+					}},
+				},
+			})
+
+			expectedListeners = append(expectedListeners, v1beta1.ListenerStatus{
+				Name: v1beta1.SectionName(strconv.Itoa(port)),
+				SupportedKinds: []v1beta1.RouteGroupKind{{
+					Group: (*v1beta1.Group)(&v1beta1.GroupVersion.Group),
+					Kind:  v1beta1.Kind("HTTPRoute"),
+				}},
+				Conditions: []metav1.Condition{{
+					Type:   string(v1beta1.ListenerConditionAccepted),
+					Status: metav1.ConditionTrue,
+					Reason: "", //any reason
+				}, {
+					Type:   string(v1beta1.ListenerConditionResolvedRefs),
+					Status: metav1.ConditionTrue,
+					Reason: "", //any reason
+				}},
+				AttachedRoutes: 1,
+			})
+		}
+
+		kubernetes.NamespacesMustBeReady(t, s.Client, s.TimeoutConfig, namespaces)
+		certBytes, keyBytes, err := GetTLSSecret(s.Client, certNN)
+		require.NoErrorf(t, err, "error getting certificate: %v", err)
+
+		t.Run("should be able to add multiple HTTP listeners with dynamic ports that then becomes available for routing traffic", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+			defer cancel()
+
+			original := &v1beta1.Gateway{}
+			err := s.Client.Get(ctx, gwNN, original)
+			require.NoErrorf(t, err, "error getting Gateway: %v", err)
+
+			// verify that the implementation is tracking the most recent resource changes
+			kubernetes.GatewayMustHaveLatestConditions(t, s.TimeoutConfig, original)
+
+			mutate := original.DeepCopy()
+
+			mutate.Spec.Listeners = append(mutate.Spec.Listeners, listeners...)
+
+			err = s.Client.Patch(ctx, mutate, client.MergeFrom(original))
+			require.NoErrorf(t, err, "error patching the Gateway: %v", err)
+
+			kubernetes.GatewayStatusMustHaveListeners(t, s.Client, s.TimeoutConfig, gwNN, expectedListeners)
+
+			gwAddr, err := kubernetes.WaitForGatewayAddress(t, s.Client, s.TimeoutConfig, gwNN)
+			require.NoErrorf(t, err, "timed out waiting for Gateway address to be assigned")
+
+			for _, listener := range mutate.Spec.Listeners {
+				timeoutConfig := s.TimeoutConfig
+				if timeoutConfig.MaxTimeToConsistency < 2*time.Minute {
+					timeoutConfig.MaxTimeToConsistency = 2 * time.Minute
+				}
```

Comment:
This feels problematic - why are we overriding this manually here?

Reply:
By default `MaxTimeToConsistency` is set to 30 seconds - which was too low when I tested against Istio.

I could add a new param.

Reply:
It seems like you should be using a different timeout here, maybe this one:
https://github.com/kubernetes-sigs/gateway-api/blob/a124c94771951bf1e96556ed18071385e6babffb/conformance/utils/config/timeout.go#L38-L40

Reply:
So the issue here is the existing helper method to make HTTP(s) requests is only using `MaxTimeToConsistency`

I made two changes to maybe make this more paltable in the short term
- changed the `2*time.Minute` to be `GatewayStatusMustHaveListeners`
- ensure we wait for the Programmed=True condition prior to invoking the HTTP(s) requests